### PR TITLE
Respect goversion build directives.

### PR DESCRIPTION
**Pull request by CannibalVox**
Yesterday afternoon, context rolled out a pre-1.7 update that uses build directives to control whether the 1.6 or 1.7 files are compiled.  These are in the format +build go1.7 and +build !go1.7.  Godep does not currently respect these build directives and chokes trying to vendor context.  This should repair this behavior, allowing us to use context again.

### CODE REVIEW

**Comment by CannibalVox** [comment]
Related to #448

File: list.go [hunk 1]
```diff
@@ -216,6 +217,10 @@ func fillPackage(p *build.Package) error {
 	if p.Goroot {
 		return nil
 	}
+	version, err := goVersion()
```
Comment (freeformz) [inline list.go]: I think this should use the recorded go version from the Godeps.json file, not whichever go version is currently running.

File: list.go [hunk 2]
```diff
@@ -271,6 +276,18 @@ NextFile:
 							p.IgnoredGoFiles = append(p.IgnoredGoFiles, fname)
 							continue NextFile
 						}
+						matchesPositive, err := regexp.MatchString("go\\d+\\.\\d+", b)
```
Comment (freeformz) [inline list.go]: compile this regex at the package level in a variable and re-use maybe? allows this to be written nicer/shorter at least.

File: list.go [hunk 3]
```diff
@@ -271,6 +276,18 @@ NextFile:
...
+						if err != nil {
+							return err
+						}
+						if matchesPositive && b != version {
```
Comment (freeformz) [inline list.go]: Version build tags match the declared build tag version and newer. eg: a file with the `go1.5` tag matches go1.5 and go1.6 and go1.7, etc.

File: list.go [hunk 4]
```diff
@@ -271,6 +276,18 @@ NextFile:
...
+							p.IgnoredGoFiles = append(p.IgnoredGoFiles, fname)
+							continue NextFile
+						}
+						if b == ("!" + version) {
```
Comment (freeformz) [inline list.go]: I *think* negatives also work the same way. eg. a tag of `!go1.5` means NOT go1.5 OR go1.6 or go1.7, etc.

**Comment by freeformz** [comment]
Closing in favor of #451 for now.

Thanks for taking the time to take a stab at this though.

**Pull request closed without merging**